### PR TITLE
feat(dedicated-cloud): allow to delete at expiration

### PR DESCRIPTION
**Pull request by marie-j**
| Question         | Answer
| ---------------- | ---
| Branch?          | `fix/hpc-us`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-5532
| License          | BSD 3-Clause

## Description

 allow to delete at expiration

## Related
- [x] 2API

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/terminate/dedicatedCloud-terminate.html [hunk 1]
```diff
@@ -12,7 +12,28 @@
...
-        <p data-translate="dedicatedCloud_close_service_desc"></p>
-        <p data-translate="dedicatedCloud_close_service_question"></p>
+        <form
+            data-ng-if="!$ctrl.currentUser.isEnterprise"
+            class="mt-4"
+            name="cancelSubscriptionForm"
+            novalidate
+        >
+            <oui-field
+                label="{{:: 'server_close_service_form_label' | translate }}"
```
Comment (radireddy) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/terminate/dedicatedCloud-terminate.html:22]:
```suggestion
               data-label="{{:: 'server_close_service_form_label' | translate }}"
```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/terminate/dedicatedCloud-terminate.html [hunk 2]
```diff
@@ -12,7 +12,28 @@
...
+            >
+                <oui-select
+                    required
+                    items="$ctrl.TERMINATE_OPTIONS"
```
Comment (radireddy) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/terminate/dedicatedCloud-terminate.html:26]:
```suggestion
                    data-items="$ctrl.TERMINATE_OPTIONS"
```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/terminate/dedicatedCloud-terminate.html [hunk 3]
```diff
@@ -12,7 +12,28 @@
...
+                    model="$ctrl.model.termination"
+                    match="label"
```
Comment (radireddy) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/terminate/dedicatedCloud-terminate.html:28]:
```suggestion
                    data-model="$ctrl.model.termination"
                    data-match="label"
```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/terminate/dedicatedCloud-terminate.routes.js [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
+    views: {
+      modal: {
+        component: 'dedicatedCloudTerminate',
+      },
+    },
+    layout: 'modal',
+    resolve: {
+      serviceInfos: /* @ngInject */ ($http, serviceName) =>
+        $http
+          .get(`/dedicatedCloud/${serviceName}/serviceInfos`)
```
Comment (radireddy) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/terminate/dedicatedCloud-terminate.routes.js:13]: caching not required?

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/terminate/dedicatedCloud-terminate.html (hunk 1 above)
Comment (antleblanc) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/terminate/dedicatedCloud-terminate.html:22]: @radireddy I've noticed on some code reviews that you're pointing that there are some missing `data-` prefix.

We are shifting on this behavior. Only for webcomponents, meaning that on `<oui-field>`, we are ok to stop using them.

Hope you'll find this useful.
Thank you!

**Review by antleblanc — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/terminate/dedicatedCloud-terminate.routes.js (hunk 4 above)
Comment (marie-j) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/terminate/dedicatedCloud-terminate.routes.js:13]: We will cache to potentially remove it as we are most likely to PUT on the same route, not sure it is worth caching

**Review by antleblanc — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/app.js [hunk 5]
```diff
@@ -319,6 +323,9 @@ angular
     /* @ngInject */ ($translate, $transitions) => {
       $transitions.onBefore({ to: 'app.**' }, () => $translate.refresh());
     },
-  );
+  )
+  .config((ovhFeatureFlippingProvider) => {
```
Comment (antleblanc) [inline packages/manager/apps/dedicated/client/app/app.js:327]: **thought:** For consistency
```suggestion
  .config(
    /* @ngInject */ (ovhFeatureFlippingProvider) => {
      ovhFeatureFlippingProvider.setApplicationName('dedicated');
    },
  );

```

**Review by JeremyDec — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build#14617.0 ✘  
* 
  * Build ✘

**Pull request merged**